### PR TITLE
Enabling ILB/ELB on windows using per-node, per-network LB endpoint.

### PR DESCRIPTION
**Pull request by pradipd**
This is 1 of 3 PRs (the other 2 are in swarmkit and moby repos).

This change enables ILB/ELB functionality on windows using a per-node, per-network LB endpoint.  The per-node, per-network LB endpoint was discussed with docker 
 as a solution to https://github.com/moby/moby/issues/30820.  Note that only windows is using the LB endpoint.  We have not changed the linux networking code to take advantage of the LB endpoint.

Signed-off-by: Pradip Dhara <pradipd@microsoft.com>

### CODE REVIEW

**Comment by pradipd** [comment]
@mavenugo @fcrisciani @msabansal

**Comment by fcrisciani** [comment]
@pradipd looks like there is a compilation error:
```
./service_windows.go:72: too many arguments in call to hcsshim.AddLoadBalancer
	have ([]hcsshim.HNSEndpoint, bool, string, string, number, number, number)
	want ([]hcsshim.HNSEndpoint, bool, string, uint16, uint16, uint16)
./service_windows.go:106: too many arguments in call to hcsshim.AddLoadBalancer
	have ([]hcsshim.HNSEndpoint, bool, string, string, uint16, uint16, uint16)
	want ([]hcsshim.HNSEndpoint, bool, string, uint16, uint16, uint16)

```

**Comment by pradipd** [comment]
I needed to vendor the latest hcsshim.
How do I build libnetwork?

**Comment by fcrisciani** [comment]
@pradipd you can just do `make` but that after the build will also run the tests, you can then just ctrl+c

File: endpoint.go [hunk 1]
```diff
@@ -1013,6 +1020,13 @@ func CreateOptionMyAlias(alias string) EndpointOption {
 	}
 }
 
+//CreateOptionLoadBalancer function returns an option setter for denoting the endpoint is a load balancer for a network
```
Comment (fcrisciani) [inline endpoint.go]: space missing after the `//`

File: sandbox.go [hunk 2]
```diff
@@ -916,6 +916,13 @@ func (sb *sandbox) clearNetworkResources(origEp *endpoint) error {
 			break
 		}
 	}
+
+	if index == -1 {
+		logrus.Errorf("Endpoint %s has already been deleted", ep.Name())
```
Comment (fcrisciani) [inline sandbox.go]: Considering that no action is taken, should we move this to warning?

File: service_windows.go [hunk 3]
```diff
@@ -1,11 +1,145 @@
...
+			if policies.elb != nil {
+				policies.elb.Delete()
+				policies.elb = nil
+			}
+			delete(lbPolicylistMap, lb)
+		}
+
+		ilbPolicy, err := hcsshim.AddLoadBalancer(endpoints, true, sourceVIP, vip.String(), 0, 0, 0)
+		if err != nil {
+			logrus.Errorf("Failed to add ILB policy for service %s (%s) with endpoints %v using load balancer IP %s on network %s: %v",
```
Comment (fcrisciani) [inline service_windows.go]: same here, is this an error or a warning? Also do you want to continue with the following logic or return?

**Comment by fcrisciani** [comment]
LGTM
@vieux @mavenugo @abhinandanpb can you guys take a look too?

**Comment by vieux** [comment]
LGTM

**Pull request merged**